### ISSUE TITLE
Question about Training Resnet18 for manufacturing defect classification 

### ISSUE THREAD

**Issue by estratos** [dusty-nv/jetson-inference#977]
My aproach is to retrain restnet18 model with a custom dataset of images of a plastic injected part.
 The goal is to build a model that can classify defects over a good part.

 In the Train Folder I have:
 
 Train  
            background
            defect-1
            defect-2
            defect-3
            good-part
            
 Pictures where taken with a common background  512 x 512 pix with the part centered on each image.
 I begin with Each class with about 100 pics for training, 20 in val and 3 or 4 in test.
 
 I retrained the model using train.py script, then export to onnx
 and with the imageNet example running the inference 
 
 The problem I have I don't have consistent results using usb or a mipi camera.
 
 I do have erratic results even with the common background.
 
  Is my aproach correct?

**Comment by h3retek** [comment]
Here's my two cents:

[Provided below is an 800MB dataset that includes 5000 training images, 1000 validation images, and 200 test images, each evenly split between the cat and dog classes.](https://github.com/dusty-nv/jetson-inference/blob/master/docs/pytorch-cat-dog.md)

**Comment by dusty-nv** [comment]
As @h3retek suggested, you probably need more images in your dataset.  Also if your dataset is that small, you may just want to combine/duplicate the train/val/test sets to get more data.  And the accuracy of only 20 images is not that indicative.  What accuracy did your model train to?   You can also train it for more epochs.

Also, if you re-train your model and export it to ONNX again, be sure to delete the `.engine` file in your model's folder.  Otherwise TensorRT will keep using your old model.  Deleting the `.engine` file will have TensorRT re-generate it based on your latest ONNX model.
